### PR TITLE
Kubernetes Plugin: Add podTableColumns prop

### PR DESCRIPTION
**Pull request by su-gupta**
Signed-off-by: su-gupta <sugupta@brex.com>

## Hey, I just made a Pull Request!

Export a CustomisationProps object from the k8s plugin, which includes a customPodTableColumns argument, to allow users to add additional custom columns to the Pod Table in the Deployments Accordion. I.e. the table shown belo
Screen Shot 2021-12-13 at 10 21 12 AM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: ADOPTERS.md
Comment (jrusso1020) [inline ADOPTERS.md:75]: we probably want to omit this change although it appears to just be a spacing fix?

File: plugins/kubernetes/src/components/DeploymentsAccordions/DeploymentsAccordions.tsx
Comment (jrusso1020) [inline plugins/kubernetes/src/components/DeploymentsAccordions/DeploymentsAccordions.tsx:177]: should `extraColumns` and `podTableColumns` be the same thing? Similar to the catalog which just has one prop passed in which you can override completely (recreating the default columns as you find necessary)

either that or should we merge the two objects and just continue to pass in `extraColumns` as a single prop?

**Review by jrusso1020 — COMMENTED** [review]
will let the maintainers chime in but left a couple of coments

Also curious if we should allow users to supply actions https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx#L165 similar to the CatalogTable for our future work which involves adding similar functionality to the k8s plugin

but that could be a followup

**Comment by mclarke47** [comment]
I am making some changes to pod columns as a part of https://github.com/backstage/backstage/pull/8248 to add resources and due to a regression in the Argo Rollouts component. It would be good if we aligned!

**Comment by su-gupta** [comment]
> I am making some changes to pod columns as a part of #8248 to add resources and due to a regression in the Argo Rollouts component. It would be good if we aligned!

Ah interesting looking forward to those changes! Do you have an ETA on when those will be merged? Looking at your changes I'm thinking that for this change it might be best to allow users to add additional columns instead of specifying them all from scratch and pass those in alongside the others in the extraColumns prop, and I think it would be easiest for me to just add that on top of your changes after they're merged.

**Comment by su-gupta** [comment]
@mclarke47 I've made some changes to make this more compatible with your upcoming changes
- Rename from podTableColumns to customPodTableColumns for clarity that these will be additional columns and not replace them all
- Pass through customPodTableColumns together with DeploymentColumns in DeploymentAccordion so that we only pass one ExtraColumns prop into PodsTable
- Removed columns.tsx file which made the TableColumns functions

After these changes I think we should be aligned and it should be easy to merge your changes on top of these, but let me know if there's anything else you'd like me to change!

File: ADOPTERS.md
Comment (su-gupta) [reply ADOPTERS.md:75]: Hm this file is changed when I commit by the automatic linting so it's not letting me revert it back and commit the old version locally - is there some way to turn off the auto-linting when I commit?

File: plugins/kubernetes/src/components/DeploymentsAccordions/DeploymentsAccordions.tsx
Comment (su-gupta) [reply plugins/kubernetes/src/components/DeploymentsAccordions/DeploymentsAccordions.tsx]: See convo below, merged them and passing them as a single prop now

File: plugins/kubernetes/src/components/Cluster/Cluster.tsx
Comment (mclarke47) [inline plugins/kubernetes/src/components/Cluster/Cluster.tsx:114]: Could we wrap customPodTableColumns in a Customisation object which contains info like this?

This way if we want to allow something else to be customisable we don't have to add it to all the component constructors again.

Then maybe replace it with a react context or something

**Comment by mclarke47** [comment]
Looking good so far, I've merged my changes, so you should be able to merge them in and build on top of them!

**Comment by su-gupta** [comment]
Hey @mclarke47 ! Have updated it to be compatible with your changes and added a CustomisationProps object, take a look and let me know if there's any other changes that should be added

**Review by mclarke47 — APPROVED** [review]
lgtm

**Comment by freben** [comment]
@su-gupta still some build errors on this one.

**Comment by su-gupta** [comment]
@freben Sorry this PR got a little messy after I forgot to sign a commit - created an updated PR here: https://github.com/backstage/backstage/pull/8599 to address the build failures

**Pull request closed without merging**
